src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+package Duke;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import Duke.task.Deadline;
+import Duke.task.Event;
+import Duke.task.Task;
+import Duke.exception.DukeException;
+
+public class Duke {
+	public static void main (String[] args) {
+		printStart ();
+		Scanner scan = new Scanner (System.in);
+		String input = scan.nextLine ();
+		ArrayList<Task> tasks = fileIO.readFile ();
+		int count = fileIO.getCount ();
+
+		while (!("bye".equalsIgnoreCase (input)) && !(input.isEmpty ())) {
+			try {
+				count = checkInput (input, tasks, count);
+			} catch (DukeException de) {
+				System.out.println (de.getMessage ());
+				System.out.println ("Please enter again: ");
+				printLine ();
+				continue;
+			} finally {
+				input = scan.nextLine ();
+			}
+		}
+		printEnd ();
+	}
+
+	public static int checkInput (String input, ArrayList<Task> tasks, int count) throws DukeException {
+		if (input.length () > 4) {
+			if (input.startsWith ("mark") || input.startsWith ("unmark")) {
+				changeStatus (input, tasks);
+				fileIO.writeFile (tasks);
+			} else if (input.startsWith ("todo") || input.startsWith ("deadline") || input.startsWith ("event")) {
+				count = addTask (input, tasks, count);
+				fileIO.writeFile (tasks);
+			} else if (input.startsWith ("delete") && !("delete".equalsIgnoreCase (input))) {
+				count = deleteTask (input, tasks, count);
+				fileIO.writeFile (tasks);
+			} else {
+				printLine ();
+				throw new DukeException ("OOPS!!! The description of a " + input + " cannot be empty.");
+			}
+		} else {
+			if ("list".equalsIgnoreCase (input)) {
+				listTasks (tasks);
+			} else if ("mark".equalsIgnoreCase (input) || "unmark".equalsIgnoreCase (input) || "todo".equalsIgnoreCase (input) || "event".equalsIgnoreCase (input) || "deadline".equalsIgnoreCase (input) || "delete".equalsIgnoreCase (input)) {
+				printLine ();
+				throw new DukeException ("OOPS!!! The description of a " + input + " cannot be empty.");
+			} else {
+				printLine ();
+				throw new DukeException ("OOPS!!! I'm sorry, but I don't know what that means :-(");
+			}
+		}
+		return count;
+	}
+
+	public static int addTask (String input, ArrayList<Task> tasks, int count) {
+		String[] arrInput = input.split (" ", 2);
+		printLine ();
+		System.out.println ("Got it. I've added this task:");
+
+		if ("todo".equalsIgnoreCase (arrInput[0])) {
+			tasks.add (new Task (arrInput[1]));
+		} else if ("deadline".equalsIgnoreCase (arrInput[0])) {
+			String[] arrTask = arrInput[1].split ("/by");
+			tasks.add (new Deadline (arrTask[0], arrTask[1]));
+		} else {
```

Comment:
best to include  an explicit `else if` and use the `else` to handle invalid inputs